homeassistant/components/apple_tv/remote.py

```diff
@@ -1,31 +1,61 @@
 """Remote control support for Apple TV."""
+
+import logging
+
 from homeassistant.components import remote
-from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+
+from .const import DOMAIN
 
-from . import ATTR_ATV, ATTR_POWER, DATA_APPLE_TV
+_LOGGER = logging.getLogger(__name__)
 
+PARALLEL_UPDATES = 0
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the Apple TV remote platform."""
-    if not discovery_info:
-        return
 
-    name = discovery_info[CONF_NAME]
-    host = discovery_info[CONF_HOST]
-    atv = hass.data[DATA_APPLE_TV][host][ATTR_ATV]
-    power = hass.data[DATA_APPLE_TV][host][ATTR_POWER]
-    async_add_entities([AppleTVRemote(atv, power, name)])
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Load Apple TV remote based on a config entry."""
+    identifier = config_entry.unique_id
+    name = config_entry.data[CONF_NAME]
+    manager = hass.data[DOMAIN][config_entry.unique_id]
+    async_add_entities([AppleTVRemote(name, identifier, manager)])
 
 
 class AppleTVRemote(remote.RemoteDevice):
     """Device that sends commands to an Apple TV."""
 
-    def __init__(self, atv, power, name):
+    def __init__(self, name, identifier, manager):
         """Initialize device."""
-        self._atv = atv
+        self.atv = None
         self._name = name
-        self._power = power
-        self._power.listeners.append(self)
+        self._identifier = identifier
+        self._manager = manager
+
+    async def async_added_to_hass(self):
+        """Handle when an entity is about to be added to Home Assistant."""
+        self._manager.listeners.append(self)
+
+    @callback
+    def device_connected(self):
+        """Handle when connection is made to device."""
+        self.atv = self._manager.atv
+
+    @callback
+    def device_disconnected(self):
+        """Handle when connection was lost to device."""
+        self.atv = None
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "identifiers": {(DOMAIN, self._identifier)},
+            "manufacturer": "Apple",
+            "model": "Remote",
```

Comment:
The model should be the same as the media player, so it is making one device for both platforms.

If you want to have a specific name for the remote, add " remote" to the name, so it can be "[Apple TV name] remote".